packages/voila/src/plugins.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  JupyterFrontEnd,
+  JupyterFrontEndPlugin
+} from '@jupyterlab/application';
+
+import { PageConfig } from '@jupyterlab/coreutils';
+
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+
+import { KernelAPI, ServerConnection } from '@jupyterlab/services';
+
+import { KernelConnection } from '@jupyterlab/services/lib/kernel/default';
+
+import { ITranslator, TranslationManager } from '@jupyterlab/translation';
+
+import {
+  IJupyterWidgetRegistry,
+  IWidgetRegistryData
+} from '@jupyter-widgets/base';
+
+import { VoilaApp } from './app';
+
+import { WidgetManager as VoilaWidgetManager } from './manager';
+
+/**
+ * The default paths.
+ */
+const paths: JupyterFrontEndPlugin<JupyterFrontEnd.IPaths> = {
+  id: '@voila-dashboards/voila:paths',
+  activate: (
+    app: JupyterFrontEnd<JupyterFrontEnd.IShell>
+  ): JupyterFrontEnd.IPaths => {
+    return (app as VoilaApp).paths;
+  },
+  autoStart: true,
+  provides: JupyterFrontEnd.IPaths
+};
+
+/**
+ * A plugin to stop polling the kernels, sessions and kernel specs.
+ *
+ * TODO: a cleaner solution would involve a custom ServiceManager to the VoilaApp
+ * to prevent the default behavior of polling the /api endpoints.
```

Comment:
This plugin is temporary and should ideally be removed.


Two options for moving forward with this:

- submit upstream fixes to the JupyterLab `ServiceManager`, for example by making some of the services optionals (maybe a `ServiceManager` shouldn't require having a kernel manager)
- reuse the work or approach done in [JupyterLite](https://github.com/jupyterlite/jupyterlite), ideally also highly configurable

Reply:
We should now be able to update to the `4.0.0-alpha.x` JupyterLab packages, which include the change from https://github.com/jupyterlab/jupyterlab/pull/11537 to allow for mock services.

Reply:
Indeed. Not sure there is more to do for now if we stick to the JupyterLab 3 packages.

A follow-up would be to update to the JupyterLab 4 packages:

> We should now be able to update to the `4.0.0-alpha.x` JupyterLab packages, which include the change from [jupyterlab/jupyterlab#11537](https://github.com/jupyterlab/jupyterlab/pull/11537) to allow for mock services.